dspace-server-webapp/src/main/java/org/dspace/app/rest/ClarinBitstreamImportController.java

```diff
@@ -0,0 +1,223 @@
+            BitstreamFormat bitstreamFormat = null;
+            if (!Objects.isNull(bitstreamFormatId)) {
+                bitstreamFormat = bitstreamFormatService.find(context, bitstreamFormatId);
+            }
+            bitstream.setFormat(context, bitstreamFormat);
+            String deletedString = request.getParameter("deleted");
+            //join created bitstream with file stored in assetstore
+            if (clarinBitstreamService.addExistingFile(context, bitstream, bitstreamRest.getSizeBytes(),
+                    bitstreamRest.getCheckSum().getValue(), bitstreamRest.getCheckSum().getCheckSumAlgorithm())) {
+                if (bitstreamRest.getMetadata().getMap().size() > 0) {
+                    metadataConverter.setMetadata(context, bitstream, bitstreamRest.getMetadata());
+                }
+                if (Boolean.parseBoolean(deletedString)) {
+                    bitstreamService.delete(context, bitstream);
+                } else {
+                    //set bitstream as primary bitstream for bundle
+                    //if bitstream is not primary bitstream, bundle is null
+                    String primaryBundleUUIDString = request.getParameter("primaryBundle_id");
+                    if (StringUtils.isNotBlank(primaryBundleUUIDString)) {
+                        UUID primaryBundleUUID = UUID.fromString(primaryBundleUUIDString);
+                        try {
+                            Bundle primaryBundle = bundleService.find(context, primaryBundleUUID);
+                            primaryBundle.setPrimaryBitstreamID(bitstream);
+                            bundleService.update(context, primaryBundle);
+                        } catch (SQLException e) {
+                            log.error("Something went wrong trying to find the Bundle with uuid: " +
+                                    primaryBundleUUID, e);
+                        }
+                    }
+                }
+                bitstreamService.update(context, bitstream);
+            } else {
+                return null;
+            }
+
+            if (bundle != null) {
+                List<Item> items = bundle.getItems();
+                if (!items.isEmpty()) {
+                    item = items.get(0);
+                }
+                if (item != null && !(authorizeService.authorizeActionBoolean(context, item, Constants.WRITE)
+                        && authorizeService.authorizeActionBoolean(context, item, Constants.ADD))) {
+                    throw new AccessDeniedException("You do not have write rights to update the Bundle's item");
+                }
+                if (item != null) {
+                    itemService.update(context, item);
+                }
+                bundleService.update(context, bundle);
+            }
+            bitstreamRest = converter.toRest(bitstream, utils.obtainProjection());
+            context.commit();
```

Comment:
refactor this `if else..`

Reply:
Which part of the code could I refactor?